<!DOCTYPE html>
<html>
<head>
<style>
body {color: Maroon;
	font-family: "arial";
	margin-left: 3cm;
	margin-right: 3cm;
	margin-top: 1cm;
	margin-bottom: 1cm;}
h1   {color: DarkSlateBlue;}
h2   {color: DarkSlateBlue;}
p    {color: Maroon;}
#nav {margin: 1cm;}
#intro {width: 80%;
	margin-top: 1cm;
	margin-bottom: 1cm;}
#obj {margin-top: 1cm;
	margin-bottom: 1cm;
	width: 95%;}
#mthd {margin-top: 1cm;
	margin-bottom: 1cm;
	width: 95%;}
#results {margin-top: 1cm;
	margin-bottom: 1cm;
	width: 95%;}
#disc {margin-top: 1cm;
	margin-bottom: 1cm;
	width: 95%;}
#future {margin-top: 1cm;
	margin-bottom: 1cm;
	width: 95%;}
#ref {margin-top: 1cm;}
a:link {color: FireBrick;
	text-decoration: none;}
a:visited {color: #460713;
	text-decoration: none;}
a:hover {color: Crimson;
	text-decoration: underline;}
a:active {color: Gold;
	text-decoration: underline;}
figure {margin-top: 1cm;
	margin-bottom: 1cm;}
</style>
<title>Machine Learning for Diagnostics</title>
</head>
<body>


<div id="title">
<h1>Neural Networks for Disease State Prediction Using Metagenomic Markers</h1>
<h4>Machine Learning, Fall 2017</h4>
<h4>Min Oh and Sheila Zhu</h4>
<hr>
</div>

<div id="nav">
<h2>Summary of Contents</h2>
<h3><ol><li><a href="#intro">Introduction</a></li>
	<li><a href="#obj">Contribution Objectives</a></li>
	<li><a href="#mthd">Data and Methodology</a></li>
	<li><a href="#results">Results</a></li>
	<li><a href="#disc">Analysis and Discussion</a></li>
	<li><a href="#future">Future Work</a></li>
	<li><a href="#ref">References</a></li>
	</ol></h3>
</div>
<br><br>


<div id="intro">
<h2>Introduction</h2>
<p>The human microbiome, which has been referred to as our second genome, consists of the microorganisms that live in and on our bodies [<a href="#r1">1</a>]. They interact with the immune system, influence metabolic activities, and consequently also affect disease state [<a href="#r2">2</a>]. Although the human microbiome is crucial for human health, it has largely been ignored. However, it has recently gained attention as a promising component of clinical diagnostics, as the substantial progress of next-generation DNA sequencing technologies has made it possible to capture a snapshot of microbial communities from human samples [<a href="#r1">1</a>, <a href="#r3">3</a>].</p>
</div>

<div id="obj">
<h2>Contribution Objectives</h2>
<p>Previous studies have proposed disease-predictive models based on high-resolution microbial features, but as they only tested SVM, RF, Lasso, and elastic net (ENet) classification methods, the application of machine learning models is still lacking elaboration [<a href="#r4">4</a>]. While focusing primarily on the additional application of neural network machine learning methods, we hoped to explore the following questions:<br>
<ul><li>What data analytic techniques are suitable for human microbiome profile-based disease prediction?</li>
	<li>Can a neural network model surpass existing methods?</li>
	<li>What real-world insights can be learned from this project?</li>
</ul>
</p>
</div>

<div id="mthd">
<h2>Data and Methodology</h2>
<p>We built off the same datasets used in the study([<a href="#r4">4</a>]), which were extracted using the open-source tool <a href="http://segatalab.cibio.unitn.it/tools/metaml/">MetAML</a> and are accessible from a public database. In addition, we explored their methodology to ensure our methodology and results could be comparable to those of the study through replicating the datasets [<a href="t1">Table 1</a>], and running preliminary tests. 
</p>
<figure id="t1"><img src="images/journal.pcbi.1004977.t001.PNG" alt="Table 1" width=95%>
<figcaption>Table 1: The number of samples per dataset. [<a href="#r4">4</a>] The T2D data was collected from Chinese women while the WT2D data was collected from European women.</figcaption></figure>
<p>The data was initially already separated by species abundance and strain-specific markers data, but included many more samples than was used in the study. In process of duplicating the datasets, we found that samples with certain disease fields were excluded from the study, presumably in order to reduce the ambiguity of the data. The reasoning behind some exclusions were obvious, such as samples with the disease state of " -" and the "n" disease state for the obesity dataset, where the foil of "obesity" was "leanness." The removal of "impaired glucose tolerance" samples in the WT2D dataset likely served to make the data binary- not only for the aforementioned purpose of reducing ambiguity, but also in order to match the binary nature of the T2D dataset. On the other hand, the removal of the "large adenoma" samples while keeping "small adenoma" in the colorectal cancer dataset is not immediately obvious, but could be related to its small sample size.
</p>
</div>

<div id="results">
<h2>Results</h2>
<p>We built classification models based on the species abundance profile for each disease data set using Random Forest and Multi-layer Perceptron algorithms implemented in <a href="https://www.cs.waikato.ac.nz/ml/weka/">Weka</a>. The data sets were split into training and test sets with ratios of 66.6 percent and 33.3 percent, respectively. We compared the classification performance of Random Forest, one of the most successful algorithms in the previous paper [<a href="#f1">Figure 1</a>], with that of the Multi-layer Perceptron [<a href="t2">Table 2</a>]. In four cases, Random Forest shows better performance. However, it did not surpass the Multi-layer Perceptron in all cases. Specifically, Neural Networks showed better performance for predicting obesity and type 2 diabetes (European cohort). 
</p>
<figure id="f1"><img src="images/journal.pcbi.1004977.g002.PNG" alt="Figure 1" width=95%>
<figcaption>Figure 1: Cross-validated results from previous study [<a href="#r4">4</a>]</figcaption></figure>
<figure id="t2"><img src="images/20171213235716.png" alt="Table 2" width=95%>
<figcaption>Figure 2: Cross-validated preliminary results comparing the prediction performance Random Forest(RF) and Multi-layer Perceptron algorithms</figcaption></figure>
<p>
After performing the preliminary experiments, we were interested to see if performing feature selection on the dataset could improve prediction performance, 
</p>
<figure id="t3"><img src="images/20171214.png" alt="Table 3" width=95%>
<figcaption>Figure 3: Performance of RF, ML Perceptron, and SVM on datasets after Feature Selection.</figcaption></figure>
</div>

<div id="disc">
<h2>Analysis and Discussion</h2>
<p>
</p>
</div>

<div id="future">
<h2>Future Work</h2>
<p>As the time frame limited what could be included in the scope of this project, there are several ways to extend on it: exploration of more hyperparameter optimization methods with cross-validation schemes and incorporating strain-specific marker data in the comparison of the 
</p>
</div>

<hr>
<div id="ref">
<h2>References</h2>
<p><ol><li id='r1'>Grice, Elizabeth A., and Julia A. Segre. "The human microbiome: our second genome." Annual review of genomics and human genetics 13 (2012): 151-170.</li>
	<li id='r2'>Honda, Kenya, and Dan R. Littman. "The microbiota in adaptive immune homeostasis and disease." Nature 535.7610 (2016): 75-84.</li>
	<li id='r3'>Hardwick, Simon A., Ira W. Deveson, and Tim R. Mercer. "Reference standards for next-generation sequencing." Nature Reviews Genetics 18.8 (2017): 473-484.</li>
	<li id='r4'>Pasolli, Edoardo, et al. <a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004977">"Machine learning meta-analysis of large metagenomic datasets: tools and biological insights."</a> PLoS computational biology 12.7 (2016): e1004977.</li>
	<li id='r5'>Truong, Duy Tin, et al. "MetaPhlAn2 for enhanced metagenomic taxonomic profiling." Nature methods 12.10 (2015): 902-903.</li>
	</ol></p>
</div>


</body>
</html>
